### PR TITLE
add config report app

### PR DESCRIPTION
**Pull request by felixboehm**
Added config_report functionality to core. Allow all admins to generate a config report.
Fix: https://github.com/owncloud/enterprise/issues/959
@DeepDiver1975 

@cdamken @michaelstingl Please test

### CODE REVIEW

**Comment by felixboehm** [comment]
- [ ] generate config report with occ command
- [ ] transifex
- [ ] review & test
- [ ] adjust build / ship scripts ??
- [ ] rewrite ajax code with a controller

File: apps/config_report/appinfo/version [hunk 1]
```diff
@@ -0,0 +1 @@
+0.1.0
```
Comment (DeepDiver1975) [inline apps/config_report/appinfo/version]: kill file

**Comment by cdamken** [comment]
- [ ] redis config array printed as array instead of object

**Comment by felixboehm** [comment]
@butonic thanks for caring here, you rock!

**Comment by butonic** [comment]
there is not much we can do about the redis array being printed as an object, because json arrays must have ordered keys. It doesn't add any value IMO.

**Comment by butonic** [comment]
@cdamken Let me try to explain why encoding the redis config as an array is impossible. Take this excerpt from the report:
```json

    "config": {
        "apps_paths": [
            {
                "path": "\/home\/jfd\/Repositories\/oc\/core\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/home\/jfd\/Repositories\/oc\/core\/apps-repos",
                "url": "\/apps-repos",
                "writable": true
            }
        ],
        "loglevel": 0,
        "wnd.logging.enable": true,
        "instanceid": "ocyx86ktd84i",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "asylum"
        ],
        "datadirectory": "\/home\/jfd\/Repositories\/oc\/core\/data",
        "overwrite.cli.url": "https:\/\/asylum\/Repositories\/oc\/core",
        "dbtype": "sqlite3",
        "version": "9.2.0.1",
        "logtimezone": "UTC",
        "installed": true
    },
```

What you expect is something like `trusted_domains`, what you get for the redis config is something like the individual `apps_paths` values. The difference is that `trusted_domains` is a numbered array, and `apps_paths` values are associative arrays. The report uses JSON. The config file uses PHP, which has a different and more complex syntax. Lot me show you the original config.php:
```php
<?php
$CONFIG = array (
  'apps_paths' => 
  array (
    0 => 
    array (
      'path' => '/home/jfd/Repositories/oc/core/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 => 
    array (
      'path' => '/home/jfd/Repositories/oc/core/apps-repos',
      'url' => '/apps-repos',
      'writable' => true,
    ),
  ),
  'loglevel' => 0,
  'wnd.logging.enable' => true,
  'instanceid' => 'ocyx86ktd84i',
  'passwordsalt' => '+xChzeXJbWB2SpAhZqZNHuf1RDmdwl',
  'secret' => 'W8DptwZ9CJDnyoOewxCzVO8Z/7Kvz4eLMbuyY2gpKFUzCEmE',
  'trusted_domains' => 
  array (
    0 => 'asylum',
  ),
  'datadirectory' => '/home/jfd/Repositories/oc/core/data',
  'overwrite.cli.url' => 'https://asylum/Repositories/oc/core',
  'dbtype' => 'sqlite3',
  'version' => '9.2.0.1',
  'logtimezone' => 'UTC',
  'installed' => true,
);
```
Whenever a numbered array (eg `array(0=>'a',1=>'b')`) like `trusted_domains` is converted to json  the keys are dropped (they don't add meaningful information). That is not possible when using an associative array (eg `array('foo'=>'a','bar'=>'b')`). To represent that as json an object is necessary.

I hope that explains it. Ping me if you want some more explanation on this.

**Comment by butonic** [comment]
shouldn't this live in a separate repository for the app?

**Comment by PVince81** [comment]
So, who will create the new repo and set it up with travis, etc ?

**Comment by butonic** [comment]
@PhilippSchaffrath please move to https://github.com/owncloud/configreport and add an occ command, also see https://github.com/owncloud/core/pull/25696#issuecomment-237301742

**Comment by butonic** [comment]
@michaelstingl how is https://github.com/michaelstingl/owncloudconf-logs related?

**Comment by michaelstingl** [comment]
> how is https://github.com/michaelstingl/owncloudconf-logs related?

This is the talk I had at the conference. I opened the issue in my repo to remind me to add the config_report to my slides. No action required.

**Comment by DeepDiver1975** [comment]
in its own repo now -> kill

**Pull request closed without merging**